iree/compiler/Conversion/Common/LinalgBufferizePass.cpp

```diff
@@ -303,6 +308,58 @@ static LogicalResult convertSubTensorOp(
       allocationDynamicSizes, bvm);
 }
 
+/// Converts a `subtensor_insert` operation to buffers by
+/// - Allocating a buffer for the result (if needed), and copying the
+///   destination value into this tensor.
+/// - Copying the source values into a subview of the result.
+static LogicalResult convertSubTensorInsertOp(
+    OpBuilder &b, WorkgroupMemoryAllocationFn allocationFn,
+    SubTensorInsertOp op, BlockAndValueMapping &bvm) {
+  OpBuilder::InsertionGuard g(b);
+  b.setInsertionPoint(op);
+  Location loc = op.getLoc();
+  Value dest = op.dest();
+  Value inputBuffer = bvm.lookup(dest);
+  auto allocationDynamicSizes = llvm::to_vector<4>(llvm::map_range(
+      llvm::seq<unsigned>(0,
+                          inputBuffer.getType().cast<ShapedType>().getRank()),
+      [&](int64_t dim) {
+        return b.createOrFold<DimOp>(loc, inputBuffer, dim);
+      }));
```

Comment:
[optional] I really don't like this style. I feel the below is cleaner and simpler:

```
  SmallVector<Value, 4> dynSizes;
  int rank = inputBuffer.getType().cast<ShapedType>().getRank();
  for (int i = 0; i < rank; ++i) {
    dynSizes.push_back(b.createOrFold<DimOp>(loc, inputBuffer, d));
  }
```

It only has five lines, and it is cleaner to me.

Reply:
+1.

Reply:
I hate writing loops now, but here you do have a point.